praxis_data/lib/repositories/jokes/data_jokes_repository.dart

```diff
@@ -1,34 +1,49 @@
 import 'dart:async';
 
+import 'package:flutter/foundation.dart';
+import 'package:get_it/get_it.dart';
 import 'package:injectable/injectable.dart';
-import 'package:logging/logging.dart';
-import 'package:praxis_data/mapper/entity_mapper.dart';
 import 'package:praxis_data/mapper/jokes/jokes_mappers.dart';
-import 'package:praxis_data/models/jokes/dt_joke_list.dart';
-import 'package:praxis_data/network/http_helper.dart';
-import 'package:praxis_data/network/url.dart';
-import 'package:praxis_flutter_domain/repositories/jokes/jokes_repository.dart';
+import 'package:praxis_data/sources/local/praxis_database.dart';
+import 'package:praxis_data/sources/network/jokes_api.dart';
+import 'package:praxis_flutter_domain/entities/api_response.dart';
 import 'package:praxis_flutter_domain/entities/jokes/dm_joke_list.dart';
+import 'package:praxis_flutter_domain/repositories/jokes/jokes_repository.dart';
 
 @Injectable(as: JokesRepository)
 class DataJokesRepository implements JokesRepository {
-  final Logger _logger = Logger('DataJokesRepository');
-
-  JokesListMapper mapper;
+  final JokesListMapper mapper = GetIt.I.get();
+  final JokesApi _jokesApi = GetIt.I.get();
+  final PraxisDatabase _praxisDatabase = GetIt.I.get();
+  final JokeMapper _jokeMapper = GetIt.I.get();
 
-  DataJokesRepository(this.mapper);
+  DataJokesRepository();
 
   @override
-  Future<DMJokeList> getFiveRandomJokes() async {
-    Map<String, dynamic> body;
-    try {
-      Uri uri = Uri.https(URL.baseUrl, URL.fiveRandomJokesPath);
-      body = await HttpHelper.invokeHttp(uri, RequestType.get);
-    } catch (error) {
-      _logger.warning('Could not get Jokes', error);
-      rethrow;
+  Future<ApiResponse<DMJokeList>> getFiveRandomJokes() async {
+    String? type;
+    // Checking platform since sqflite doesn't support WEB
+    if (defaultTargetPlatform == TargetPlatform.android || defaultTargetPlatform == TargetPlatform.iOS) {
+      try {
+        final networkResponse = await _jokesApi.getFiveRandomJokes();
+        if (networkResponse is Success) {
+          await _praxisDatabase.deleteAllJokes();
+          final networkJokes = (networkResponse as Success).data as DMJokeList;
+          final jokes = mapper.mapToData(networkJokes);
```

Comment:
can we have separate data sources for local and network @shubhamsinghmutualmobile ?

Reply:
What I understand by 'separate data sources' is that we must have two different classes from where the required data would be coming from. In this case, we have PraxisDatabase for Local and JokesApi for Network data.

Exactly what changes are expected here?

Reply:
We need to have a JokesLocalDataSource and JokesNetworkDataSource